Doc/library/datetime.rst

```diff
@@ -1228,27 +1380,40 @@ Instance methods:
 
 .. method:: datetime.isocalendar()
 
-   Return a 3-tuple, (ISO year, ISO week number, ISO weekday).  The same as
+   Return a 3-tuple, (ISO year, ISO week number, ISO weekday). The same as
    ``self.date().isocalendar()``.
 
 
 .. method:: datetime.isoformat(sep='T', timespec='auto')
 
-   Return a string representing the date and time in ISO 8601 format,
-   YYYY-MM-DDTHH:MM:SS.ffffff or, if :attr:`microsecond` is 0,
-   YYYY-MM-DDTHH:MM:SS
+   Return a string representing the date and time in ISO 8601 format:
+
+   - ``YYYY-MM-DDTHH:MM:SS.ffffff``, if :attr:`microsecond` is not 0
+   - ``YYYY-MM-DDTHH:MM:SS``, if :attr:`microsecond` is 0
 
    If :meth:`utcoffset` does not return ``None``, a string is
    appended, giving the UTC offset:
-   YYYY-MM-DDTHH:MM:SS.ffffff+HH:MM[:SS[.ffffff]] or, if :attr:`microsecond`
-   is 0 YYYY-MM-DDTHH:MM:SS+HH:MM[:SS[.ffffff]].
+
+   - ``YYYY-MM-DDTHH:MM:SS.ffffff+HH:MM[:SS[.ffffff]]``, if :attr:`microsecond`
+     is not 0
+   - ``YYYY-MM-DDTHH:MM:SS+HH:MM[:SS[.ffffff]]``,  if :attr:`microsecond` is 0
+
+   Examples::
+
+       >>> from datetime import datetime, timezone
+       >>> datetime(2019, 5, 18, 15, 17, 8, 132263).isoformat()
+       '2019-05-18T15:17:08.132263'
+       >>> datetime(2019, 5, 18, 15, 17, tzinfo=timezone.utc).isoformat()
+       '2019-05-18T15:17:00+00:00'
 
    The optional argument *sep* (default ``'T'``) is a one-character separator,
-   placed between the date and time portions of the result.  For example,
+   placed between the date and time portions of the result. For example::
 
       >>> from datetime import tzinfo, timedelta, datetime
       >>> class TZ(tzinfo):
-      ...     def utcoffset(self, dt): return timedelta(minutes=-399)
+      ...     """A time zone with an arbitrary, constant -06:39 offset."""
```

Comment:
Maybe add seconds and/or or microseconds here? None of the other examples include them, but they have been supported since Python 3.6.

Reply:
Added in e116e8e880; let me know if I was misinterpreting your suggestion.